### PR TITLE
[installer]: configure and validate the external database

### PR DESCRIPTION
**Pull request by MrSimonEmms**
## Description

Enables use of external MySQL DB. This is necessary for AWS/Azure and other cloud providers

## Related Issue(s)

Fixes https://github.com/gitpod-io/gitpod-microsoft-aks-guide/issues/5

## Release Notes

```release-note
Configure and validate the external database
```

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **mrsimonemms** after the PR has been reviewed.

Associated issue: *#5*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[installer/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/installer/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: installer/pkg/components/database/external/init/00-create-and-init-sessions-db.sql [hunk 1]
```diff
@@ -0,0 +1,16 @@
+-- Copyright (c) 2020 Gitpod GmbH. All rights reserved.
+-- Licensed under the MIT License. See License-MIT.txt in the project root for license information.
+
+-- must be idempotent
+
+CREATE DATABASE IF NOT EXISTS `gitpod-sessions` CHARSET utf8mb4;
+
+USE `gitpod-sessions`;
+
+CREATE TABLE IF NOT EXISTS sessions (
```
Comment (csweichel) [inline installer/pkg/components/database/external/init/00-create-and-init-sessions-db.sql:10]: we might want to extra those init files to a `pkg/database/init` package.

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **csweichel** after the PR has been reviewed.

Associated issue: *#5*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[installer/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/installer/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: installer/pkg/components/database/external/init/00-create-and-init-sessions-db.sql (hunk 1 above)
Comment (MrSimonEmms) [reply installer/pkg/components/database/external/init/00-create-and-init-sessions-db.sql]: I've exported the `external` and `cloudsql` init to a `pkg/database/init` package. As the `incluster` init scripts are subtly different, I've left those out but we may want to incorporate that in future

**Comment by csweichel** [comment]
did not run this, but code lgtm

/lgtm

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: e2f8de62d7ba7ba2f094de456710789f25ecd4a1</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/6716#issuecomment-972755101" title="LGTM">csweichel</a>*

Associated issue: *#5*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[installer/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/installer/OWNERS)~~ [csweichel]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**